src/Accelerators/NNPA/Conversion/ONNXToZHigh/DevicePlacementHeuristic.hpp

```diff
@@ -0,0 +1,66 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===-------- DevicePlacementHeuristic.hpp - Place ops using model  -------===//
+//
+// Copyright 2023 The IBM Research Authors.
+//
+// =============================================================================
+//
+// This file contains heuristics to place operations on CPU or NNPA.
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "mlir/IR/BuiltinOps.h"
+
+#include "src/Dialect/ONNX/ONNXDimAnalysis.hpp"
+
+namespace onnx_mlir {
+
+using OpSetType = mlir::DenseSet<mlir::Operation *>;
+
+// Place all ops that qualify for NNPA executions on the NNPA.
+void PlaceAllLegalOpsOnNNPA(mlir::MLIRContext *context,
+    const llvm::SmallVector<mlir::Operation *, 32> &ops,
+    const OpSetType &cpuOps);
+
+// Place all ops that qualify for NNPA execution on the NNPA when the operations
+// are estimated run faster on the NNPA.
+void PlaceBeneficialOpsOnNNPA(mlir::MLIRContext *context,
+    const llvm::SmallVector<mlir::Operation *, 32> &ops,
+    const DimAnalysis *dimAnalysis, const OpSetType &cpuOps);
+
+// Place all ops that qualify for NNPA execution on the NNPA when the operations
+// are estimated run faster on the NNPA, including the costs of Stick and
+// Unstick necessary for NNPA execution. The algorithm starts to place on the
+// CPU/NNPA operations that are significantly faster on CPU/NNPA. Then it aims
+// to add operations to the NNPA when the new operations are faster including
+// the additional (if any) stick/unstick required for these less significantly
+// faster NNPA operations. Three factors below can modify the sensitivity at
+// which ops are assigned to the NNPA.
+
+// minFactor: NNPA has to be at least minFactor times faster than CPU.
```

Comment:
It's a bit difficult to know the difference between `minFactor` and  `significantNNPAFactor`.
As I understand, we use `minFactor` in `nnpaTimeWithOverheads * minFactor <= cpuTime`, so it would be better to say ` minFactor: NNPA (including stick/untick) has to be at least minFactor times faster than CPU.`

Reply:
thanks for the suggested wording, much appreciated.